app/Api/V1/Controllers/IncidentTypeController.php

```diff
@@ -48,4 +64,271 @@ public function index()
             'data' => IncidentType::get()
         ], 200);
     }
+
+    /**
+     * @SWG\POST(
+     *     path="/api/incidents/types",
+     *     summary="Create a new incident type",
+     *     tags={"IncidentTypes"},
+     *     description="Create a new incident type.",
+     *     operationId="storeIncidentType",
+     *     consumes={"application/json"},
+     *     produces={"application/json"},
+     *     @SWG\Parameter(
+     *          in="body",
+     *          name="body",
+     *          required=true,
+     *          @SWG\Schema(
+     *              @SWG\Property(
+     *                  property="name",
+     *                  type="string"
+     *              ),
+     *              @SWG\Property(
+     *                  property="label",
+     *                  type="string"
+     *              ),
+     *              @SWG\Property(
+     *                  property="code",
+     *                  type="string"
+     *              ),
+     *              @SWG\Property(
+     *                  property="status",
+     *                  type="string",
+     *                  enum="['Active', 'Inactive']"
+     *              )
+     *          ),
+     *     ),
+     *     @SWG\Response(
+     *         response=201,
+     *         description="successful operation",
+     *         @SWG\Schema(
+     *             @SWG\Items(ref="#/definitions/IncidentType")
+     *         )
+     *     ),
+     *     @SWG\Response(
+     *         response="422",
+     *         description="Fields are missing or code is not unique.",
+     *     ),
+     *     @SWG\Response(
+     *         response=500,
+     *         description="internal server error"
+     *     )
+     * )
+     * @param Request $request
+     * @param IncidentTypeService $incidentTypeService
+     * @return \Dingo\Api\Http\Response
+     */
+    public function store(Request $request, IncidentTypeService $incidentTypeService)
+    {
+        try {
+            /** @var IncidentType $incidentType */
+            $incidentType = $incidentTypeService->create($request->all());
+        } catch (StoreResourceFailedException $exception) {
+            throw $exception;
```

Comment:
Is there a global exception handler for controllers where this exception is processed?
If there isn't, I would rather return a response with an http internal server error or bad request code rather then propagate the exception to the calling environment.

Reply:
There is the standard one, app/Exceptions/Handler.php and I tried to keep the controller appropriate with the other controllers, for eg. PrecinctController.php:85
`if ($validator->fails()) {
            throw new StoreResourceFailedException('Could not create new precinct.', $validator->errors());
  }`.
I can't see how to generate a primary key exception but for the unique exception, the message is pretty generic and it doesn't return any details about the paths or database information. This is handled in the IncidentTypeService::validate():57 and it returns only the fields and their violations( in the standard way of Laravel)
`if ($validator->fails()) {
            throw new StoreResourceFailedException('Could not create a new incident type.', $validator->errors());
  }`

Reply:
Cool, thanks for the explanation.